File: c++/Class_and_Object/Example-1/Mobile.cpp
#include <iostream>
using namespace std;

class Mobile {
public:
    string brand;
    string model;
    double price;
    int battery;

    Mobile(string b, string m, double p, int bat) {
        brand = b;
        model = m;
        price = p;
        battery = bat;
    }

    void showDetails() {
        cout << brand << " " << model << " | Price: " << price << " | Battery: " << battery << "%" << endl;
    }

    void charge(int amount) {
        battery += amount;
        cout << "Battery charged to " << battery << "%" << endl;
    }
};

int main() {
    Mobile m1("Samsung", "Galaxy S21", 70000, 50);
    Mobile m2("Apple", "iPhone 13", 90000, 30);

    m1.showDetails();
    m1.charge(20);

    m2.showDetails();
    m2.charge(40);

    return 0;
}

File: Typescript/Class_and_Object/Example-1/Mobile.ts
class Mobile {
    constructor(
        public brand: string,
        public model: string,
        public price: number,
        public battery: number
    ) {}

    showDetails() {
        console.log(`${this.brand} ${this.model} | Price: ${this.price} | Battery: ${this.battery}%`);
    }

    charge(amount: number) {
        this.battery += amount;
        console.log(`Battery charged to ${this.battery}%`);
    }
}

let m1 = new Mobile("Samsung", "Galaxy S21", 70000, 50);
let m2 = new Mobile("Apple", "iPhone 13", 90000, 30);

m1.showDetails();
m1.charge(20);

m2.showDetails();
m2.charge(40);
